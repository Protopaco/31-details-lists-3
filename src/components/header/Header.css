@import url('../../context/variables.css');

.header{
    background-color: var(--secondary);
    height: 150px;
    display: flex;
    flex-direction: column;
    color: var(--primary);
    align-items: center;
    font-size: 3em;
    margin-top: 0px;
}

.homeButton{
    background-color: var(--primary);
    color: var(--secondary);
    margin-left: 30px;
}
.backButton{
    display: flex;
    justify-items: flex-start;
    align-items: flex-start;
    width: 100vw;
}

.text{
    margin-bottom: 30px;
}

.themeButton{
    color: var(--secondary);
    background-color: var(--primary);
}