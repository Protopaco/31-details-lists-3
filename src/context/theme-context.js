import React, { createContext, useContext, useState, useEffect } from 'react'

export const ThemeContext = createContext(null);

export const ThemeProvider = ({ children }) => {
    const [theme, setTheme] = useState('light');

    const switchTheme = () => {
        if (theme === 'light') {
            setTheme('dark')
            document.documentElement.style.setProperty('--primary', 'brown');
            document.documentElement.style.setProperty('--secondary', 'beige');
        } else {
            setTheme('light')
            document.documentElement.style.setProperty('--primary', 'beige');
            document.documentElement.style.setProperty('--secondary', 'brown');
        }
    }

    return (
        <ThemeContext.Provider value={{ theme, switchTheme }}>
            {children}
        </ThemeContext.Provider>
    )
}

export const useTheme = () => {
    const { theme, switchTheme } = useContext(ThemeContext);
    return { theme, switchTheme }
}

