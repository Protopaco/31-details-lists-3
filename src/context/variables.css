:root{
    --primary: magenta;
    --secondary: rgb(1, 186, 174);
    --tertiary: tan;
}