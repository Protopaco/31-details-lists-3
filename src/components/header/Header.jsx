import React from 'react'
import styles from './Header.css'
import { Link } from 'react-router-dom'
import { useTheme } from '../../context/theme-context'

const Header = () => {
    const { theme, switchTheme } = useTheme();

    return (
        <div className={styles.header}>
            <div className={styles.backButton}>
                <Link to={'/'}>
                    <button className={styles.homeButton}>
                        HOME
                </button>
                </Link>
                <button
                    className={styles.themeButton}
                    onClick={switchTheme}>
                    {theme}
                </button>
            </div>
            <div className={styles.text}>
                FLAGS OF THE WORLD
        </div>
        </div>
    )
}

export default Header;
