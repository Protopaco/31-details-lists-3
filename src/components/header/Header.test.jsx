import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../../context/theme-context'


describe('Header component', () => {
    afterEach(() => cleanup());
    it('renders Header', () => {
        const { asFragment } = render(
            <Router>
                <ThemeProvider>
                    <Header />
                </ThemeProvider>
            </Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
