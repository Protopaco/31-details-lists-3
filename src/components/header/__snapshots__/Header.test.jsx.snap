// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="backButton"
    >
      <a
        href="/"
      >
        <button
          class="homeButton"
        >
          HOME
        </button>
      </a>
      <button
        class="themeButton"
      >
        light
      </button>
    </div>
    <div
      class="text"
    >
      FLAGS OF THE WORLD
    </div>
  </div>
</DocumentFragment>
`;
